### DISCUSSION TITLE
Feathers "proxy" app with socket.io server and client together

### DISCUSSION THREAD

**Discussion by arkabase** [feathersjs/feathers#2480] [Q&A]
For a specific project, I have 2 feathers apps : 
- A feathers remote server exposing @feathersjs/socketio
- A feathers local client talking to the server with @feathers/socketio-client
so far so good.

Now, I want to connect some local apps in react to the feathers local client, using socket.io client direct connexion. So I need to configure both @feathersjs/socketio-client and @feathers/socketio on the local feathers app, the first talking to the remote server, the last talking to the local apps.

My problem is that both the socketio and socketio-client configure packages use app.io to access their respective server/socket object !

Is there a simple way tu configure both @feathersjs/socketio-client and @feathers/socketio in the same feathers app without them to conflict one another ?

Thanks for your help

**Comment by daffl** [comment]
You would create two apps. One server app and one client app that sets up Socket.io. Note that authentication is difficult when proxying to a Socket.io endpoint because one connection is treated as a single client and not as another server handling multiple requests on behalf of other clients/users. The usual approach here is to use HTTP instead and send an authorization header along that identifies the end user making the request.
